testing/test_pdb.py

```diff
@@ -434,3 +463,115 @@ def test_foo():
 
         child.expect('custom set_trace>')
         self.flush(child)
+
+
+class TestDebuggingBreakpoints(object):
+
+    def test_supports_breakpoint_module_global(self):
+        """
+        Test that supports breakpoint global marks on Python 3.7+ and not on
+        CPython 3.5, 2.7
+        """
+        if sys.version_info.major == 3 and sys.version_info.minor >= 7:
+            assert SUPPORTS_BREAKPOINT_BUILTIN is True
+        if sys.version_info.major == 3 and sys.version_info.minor == 5:
+            assert SUPPORTS_BREAKPOINT_BUILTIN is False
+        if sys.version_info.major == 2 and sys.version_info.minor == 7:
+            assert SUPPORTS_BREAKPOINT_BUILTIN is False
+
+    @pytest.mark.skipif(not SUPPORTS_BREAKPOINT_BUILTIN, reason="Requires breakpoint() builtin")
+    def test_sys_breakpointhook_not_custom_pdb(self):
+        """
+        Test that sys.breakpointhook is not set to the custom Pdb class
+        """
+        assert sys.breakpointhook != pytestPDB.set_trace
+
+    @pytest.mark.skipif(not SUPPORTS_BREAKPOINT_BUILTIN, reason="Requires breakpoint() builtin")
+    def test_sys_breakpointhook_configure_and_unconfigure(self, testdir):
+        """
+        Test that sys.breakpointhook is set to the custom Pdb class once configured, test that
+        hook is reset to system value once pytest has been unconfigured
+        """
+        config = testdir.parseconfig()
+
+        pytest_configure(config)
+        assert sys.breakpointhook == pytestPDB.set_trace
+
+        p1 = testdir.makepyfile("""
+        def test_nothing():
+            a = 0
+            assert a == 0
+        """)
+        result = testdir.runpytest_inprocess("", p1)
+        result.stdout.fnmatch_lines([
+            "* passed*",
+        ])
+        assert sys.breakpointhook != pytestPDB.set_trace
+
+    @pytest.mark.skipif(not SUPPORTS_BREAKPOINT_BUILTIN, reason="Requires breakpoint() builtin")
+    def test_sys_breakpointhook_configure_and_unconfigure_with_pdb_flag(self, testdir):
```

Comment:
You can probably parametrize this test with the above.

```python
    @pytest.mark.skipif(not SUPPORTS_BREAKPOINT_BUILTIN, reason="Requires breakpoint() builtin")
    @pytest.mark.parametrize('args', [('--pdb',), ()])
    def test_sys_breakpointhook_configure_and_unconfigure(self, testdir, args):
        ...
        result = testdir.runpytest_inprocess(*args)
```

(It's not necessary to pass `p1` to `runpytest_inprocess`).

Reply:
Fixed in f1f4c8c1